src/header_detect_io.c

```diff
@@ -942,13 +942,13 @@ static const AMQP_HEADER sasl_amqp_header =
     sizeof(sasl_amqp_header_bytes)
 };
 
-const AMQP_HEADER header_detect_io_get_amqp_header(void)
```

Comment:
>const AMQP_HEADER [](start = 0, length = 17)

I am sorry but I am a hesitant about this. Can you tell me what was the warning that was issued? Basically header_detect_io_get_amqp_header returning a struct that is const should be perfectly fine... Unless I am missing something.

Reply:
"../../inc/azure_uamqp_c/header_detect_io.h", line 38: warning: type qualifier on return type is meaningless

[Intel](https://software.intel.com/en-us/articles/cdiag858) explains this better than I can:
"The compiler generates this warning when it finds a type qualifier applied to the return type of a function. C++ allows the type-qualifier to be const or volatile , but either qualifier applied to a function return type is meaningless, because functions can only return rvalues and the type qualifiers apply only to lvalues."

Reply:
Alright

---
In reply to: [188477256](https://github.com/Azure/azure-uamqp-c/pull/227#discussion_r188477256) [](ancestors = 188477256)